src/models/settlementWindow/facade.js

```diff
@@ -230,15 +255,44 @@ const Facade = {
+                .select('swc.settlementWindowContentId', 'pc.participantCurrencyId',
+                  knex.raw('CASE WHEN ppc.change > 0 THEN ? ELSE ?? END AS transferParticipantRoleTypeId', [ Enum.Accounts.TransferParticipantRoleType.PAYER_DFSP, Enum.Accounts.TransferParticipantRoleType.PAYEE_DFSP]),
+                  knex.raw('? AS ??', [Enum.Accounts.LedgerEntryType.PRINCIPLE_VALUE, 'ledgerEntryTypeId']),
                   knex.raw('? AS ??', [Enum.Settlements.SettlementWindowState.CLOSED, 'settlementWindowStateId']),
                   knex.raw('? AS ??', [transactionTimestamp, 'createdDate']))
-                .sum('tp.amount AS amount')
+                .sum('ppc.change AS amount')
             })
             .transacting(trx)
           await builder
 
+          const fxBuilder = knex
+            .from(knex.raw('settlementContentAggregation (settlementWindowContentId, participantCurrencyId, transferParticipantRoleTypeId, ledgerEntryTypeId, currentStateId, createdDate, amount)'))
+            .insert(function () {
+              this.from('fxTransferFulfilment AS fxtf')
+                .join('fxTransferStateChange AS fxtsc', 'fxtsc.commitRequestId', 'fxtf.commitRequestId')
+                .join('participantPositionChange AS ppc', 'ppc.fxTransferStateChangeId', 'fxtsc.fxTransferStateChangeId')
+                .join('participantCurrency AS pc', 'pc.participantCurrencyId', 'ppc.participantCurrencyId')
+                .join('participant AS p', 'p.participantId', 'pc.participantId')
+                .join('settlementWindowContent AS swc', function () {
+                  this.on('swc.settlementWindowId', 'fxtf.settlementWindowId')
+                    .on('swc.ledgerAccountTypeId', 'pc.ledgerAccountTypeId')
+                    .on('swc.currencyId', 'pc.currencyId')
+                })
+                .join('settlementModel AS m', 'm.settlementModelId', 'swc.settlementModelId')
+                .where('fxtf.settlementWindowId', settlementWindowId)
+                .andWhere('m.settlementGranularityId', Enum.Settlements.SettlementGranularity.NET)
+                .groupBy('swc.settlementWindowContentId', 'pc.participantCurrencyId', 'transferParticipantRoleTypeId', 'ledgerEntryTypeId')
+                .select('swc.settlementWindowContentId', 'pc.participantCurrencyId',
+                  knex.raw('CASE WHEN ppc.change > 0 THEN ? ELSE ?? END AS transferParticipantRoleTypeId', [ Enum.Accounts.TransferParticipantRoleType.PAYER_DFSP, Enum.Accounts.TransferParticipantRoleType.PAYEE_DFSP]),
+                  knex.raw('? AS ??', [Enum.Accounts.LedgerEntryType.PRINCIPLE_VALUE, 'ledgerEntryTypeId']),
+                  knex.raw('? AS ??', [Enum.Settlements.SettlementWindowState.CLOSED, 'settlementWindowStateId']),
+                  knex.raw('? AS ??', [transactionTimestamp, 'createdDate']))
+                .sum('ppc.change AS amount')
+            })
+            .transacting(trx)
+          await fxBuilder
+
```

Comment:
I have some concerns about having a second query that inserts into settlementContentAggregation. I suspect this table might have some intention of aggregating amounts grouped over the fields settlementWindowContentId, participantCurrencyId, transferParticipantRoleTypeId and ledgerEntryTypeId. Is this a correct statement?
If it is correct, I do not see a unique constraint for these columns and maybe we must change the 2 queries to a single insert, using union and aggregation on top of it.

Reply:
Done, thanks for spotting